contracts/test/dummy_tokens/DummyToken.sol

```diff
@@ -10,18 +10,34 @@ contract DummyToken is MintableToken {
     string public symbol;
     uint public decimals;
 
+    uint public constant TOKENS_PER_ACCOUNT = 100 * 10 ** 18;
+
     function DummyToken(
         string _name,
         string _symbol,
         uint _decimals,
-        uint _totalSupply)
+        uint _totalSupply,
+        address[] _initialBalanceHolders)
         public
     {
         name = _name;
         symbol = _symbol;
         decimals = _decimals;
         totalSupply_ = _totalSupply;
-        balances[msg.sender] = _totalSupply;
+
+        uint numInitialBalanceHolders = _initialBalanceHolders.length;
+
+        // Set the last initial balance holder's balance to the total supply
+        address supplyHolder = _initialBalanceHolders[numInitialBalanceHolders - 1];
+        balances[supplyHolder] = _totalSupply;
+
+        // For every other balance holder, distribute tokens and increase the total supply appropriately
+        for (uint i = 0; i < numInitialBalanceHolders - 1; i++) {
+            totalSupply_.add(TOKENS_PER_ACCOUNT);
```

Comment:
this feels strange to me. why increment the total supply. if i specified a total supply of 1000, i would want that respected...

Reply:
^ Agreed -- seems like it would be cleaner to allocate `totalSupply / numInitialBalanceHolders` to each of the users.  If that ends up breaking tests because the `owner` account is granted insufficient balance, then we can cross this bridge.